client/my-sites/upgrades/checkout-thank-you/google-voucher-details.jsx

```diff
@@ -0,0 +1,228 @@
+		super();
+
+		// bounds
+		this.onButtonClick = this.onButtonClick.bind( this );
+		this.onDialogCancel = this.onDialogCancel.bind( this );
+		this.onAgreeButton = this.onAgreeButton.bind( this );
+
+		// debounced
+		this.incrementStep = debounce( this.incrementStep, 300 );
+
+		this.state = {
+			step: INITIAL_STEP
+		};
+	}
+
+	componentWillMount() {
+		const voucher = this.getVoucher();
+		if ( voucher && voucher.status === 'assigned' ) {
+			this.setState( { step: CODE_REDEEMED } );
+		}
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		if ( ! nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( this.props.googleAdCredits === nextProps.googleAdCredits ) {
+			return null;
+		}
+
+		if ( nextProps.googleAdCredits.length > 0 ) {
+			this.setState( { step: CODE_REDEEMED } );
+		}
+	}
+
+	onButtonClick() {
+		this.incrementStep();
+	}
+
+	onDialogCancel() {
+		this.setState( { step: INITIAL_STEP } );
+	}
+
+	onAgreeButton() {
+		this.props.assignVoucher( this.props.selectedSite.ID, GOOGLE_CREDITS );
+
+		this.setState( { step: CODE_REDEEMED } );
+	}
+
+	incrementStep() {
+		const newStep = ( this.state.step === INITIAL_STEP ) ? TERMS_AND_CONDITIONS : CODE_REDEEMED;
+		this.setState( { step: newStep } );
+	}
+
+	getVoucher( props = this.props ) {
+		const { googleAdCredits } = props;
+		return googleAdCredits.length > 0 ? googleAdCredits[0] : {};
+	}
+
+	renderInitialStep() {
+		return (
+			<div className="purchase-detail__body">
+				<PurchaseButton
+					onClick={ this.onButtonClick }
+					text={ this.props.translate( 'Generate Code' ) } />
+
+				<TipInfo
+					info={ this.props.translate( 'Offer valid in US and Canada after spending the first $25 on Google AdWords.' ) } />
+			</div>
+		);
+	}
+
+	renderTermsAndConditions() {
+		return (
+			<Dialog
+				isVisible={ true }
+				onClose={ this.onDialogCancel }
+				additionalClassNames="google-voucher-dialog"
+			>
+				<div className="google-voucher-dialog__header">
+					<img
+						src="/calypso/images/google-vouchers/google-voucher.svg"
+						className="google-voucher-dialog__header__image" />
+
+					<div className="google-voucher-dialog__header__text">
+						<h1>{ this.props.translate( 'Terms of Service' ) }</h1>
+						<p>{ this.props.translate( 'Google AdWords credit' ) }</p>
+					</div>
+				</div>
+
+				<div className="google-voucher-dialog__body">
+					{ googleTermsAndConditions() }
+				</div>
+
+				<div className="google-voucher-dialog__footer">
+					<PurchaseButton
+						className="google-vouchers-dialog__cancel-button"
+						primary={ false }
+						onClick={ this.onDialogCancel }
+						text={ this.props.translate( 'Cancel' ) } />
+
+					<PurchaseButton
+						className="google-vouchers-dialog__agree-button"
+						onClick={ this.onAgreeButton }
+						text={ this.props.translate( 'Agree' ) } />
+				</div>
+			</Dialog>
+		);
+	}
+
+	renderCodeRedeemed() {
+		const { code } = this.getVoucher();
+		return (
+			<div className="purchase-detail__body">
```

Comment:
ClassNames are mixed in this file. Perhaps all the google voucher stuff should be moved to its own component.